mls-rs-uniffi/src/lib.rs

```diff
@@ -420,6 +420,51 @@ impl Client {
     }
 }
 
+#[derive(Clone, Debug, uniffi::Object)]
+pub struct RatchetTree {
+    inner: mls_rs::group::ExportedTree<'static>,
+}
+
+impl From<mls_rs::group::ExportedTree<'static>> for RatchetTree {
+    fn from(inner: mls_rs::group::ExportedTree<'static>) -> Self {
+        Self { inner }
+    }
+}
+
+#[uniffi::export]
+impl RatchetTree {
+    /// Encode the ratchet tree in MLS encoding.
+    pub fn to_bytes(&self) -> Result<Vec<u8>, Error> {
+        self.inner.to_bytes().map_err(Into::into)
+    }
+
+    /// Return size of ratchet tree in MLS encoding.
+    pub fn byte_size(&self) -> u64 {
+        #[cfg(not(target_pointer_width = "64"))]
+        compile_error!("mls-rs-uniffi expects a 64-bit pointer width");
+
+        self.inner.byte_size().try_into().unwrap()
+    }
```

Comment:
Out of curiosity, what does it have to do with pointer width? Can't we convert whatever `usize` is to `u64` (any tree size will fit there for sure)?

Reply:
Maybe I wrote something silly here... My intention is to do a lossless conversion of the `usize` to `u64`. However, `TryFrom` will fail if we're on a 32-bit machine and so I tried to give a better compiler error in that case.

I would really have liked to just write `self.inner.byte_size() as u64`,but the `as` conversions are frowned upon these days. On a a 64-bit machine, the [`try_into()` call ends up calling `as`](https://doc.rust-lang.org/stable/src/core/convert/num.rs.html#390).

Reply:
Why would it fail on a 32-bit machine, though? We can convert `u32` to `u64`, no?

Reply:
AFAIK, on existing platforms, this conversion never fails. If there ever are architectures with 128-bit pointers, such a conversion would become fallible and apparently that's why the `std` lib does not provide `From<usize> for u64`.

Reply:
> Why would it fail on a 32-bit machine, though? We can convert `u32` to `u64`, no?

Heh, that's true... I guess I was thinking of it backwards :smile: 

> AFAIK, on existing platforms, this conversion never fails. If there ever are architectures with 128-bit pointers, such a conversion would become fallible and apparently that's why the `std` lib does not provide `From<usize> for u64`.

Yes, that matches my understanding as well.

I think I can simplify this by just removing the `compile_error!` call then.